### PR TITLE
upcoming: [M3-8217] - Tax Id Notifications & Icon

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
New notifications for Tax Id

## Changes  🔄
- Add two new notifications when tax id is valid/invalid
- Add warning icon next to tax id in billing section
   - Changed icon to solid variant for visibility (only used in this location in the app) 

## Target release date 🗓️
7/8

## Preview 📷

https://github.com/linode/manager/assets/125309814/b692dd1b-04d3-45d3-ac17-15fc03dfdea9

| Desc  | SS   |
| ------- | ------- |
| Notifications | Screenshot 2024-06-26 at 8 56 23 AM |
| Toast | Screenshot 2024-07-10 at 12 09 48 PM |
| Event | Screenshot 2024-07-10 at 3 04 59 PM |

| Icon Before  | Icon After (ignore dark mode color changes) |
| ------- | ------- |
|Screenshot 2024-06-07 at 12 57 34 PM| Screenshot 2024-07-08 at 9 59 11 AM|

### Prerequisites
- If you want to test the events part of this, you'll need to setup DevEnv (see me about details)
- Otherwise, you can locally comment out the rendering conditions for the TooltipIcon

### Validation
- Checkout this branch in devenv
- Checkout latest develop branch of apinext in devenv
- Add a row to the TaxValidation table with format_valid set to 1, and tax_registered to yes.
- curl /account/notifications and notice the tax_id_valid notification.
- Set format_valid to 0, and curl /account/notifications again, notice tax_id_invalid notification. Try again with format_valid set to 1, and tax_registered set to no.

## Related
- https://github.com/linode/manager/pull/10664

### CODE REVIEW

File: packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/ContactInformation.tsx [hunk 1]
```diff
@@ -220,9 +228,22 @@ const ContactInformation = (props: Props) => {
...
+                  sx={{
+                    margin: 0,
+                  }}
+                >
+                  <strong>Tax ID</strong> {taxId}
+                </StyledTypography>
+                {invalidTaxId && (
+                  <TooltipIcon
+                    status="warning"
+                    style={{ paddingBottom: 0, paddingTop: 0 }}
```
Comment (hkhalil-akamai) [inline packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/ContactInformation.tsx:242]: These styles appear to have no effect

**Review by hkhalil-akamai — APPROVED** [review]
I don't have devenv set up but I mocked the notification locally and verified the notification and tooltip appear as expected and display the tax id warning.

File: packages/api-v4/src/account/types.ts [hunk 2]
```diff
@@ -257,7 +257,9 @@ export type NotificationType =
   | 'promotion'
   | 'user_email_bounce'
   | 'volume_migration_scheduled'
-  | 'volume_migration_imminent';
+  | 'volume_migration_imminent'
+  | 'tax_id_valid'
```
Comment (bnussman-akamai) [inline packages/api-v4/src/account/types.ts:261]: How exactly would `tax_id_valid` work as a notification? 

Would it just be returned all the time?

Usually notifications are only used for urgent things that need resolution.

In my opinion, `tax_id_invalid` should not exist as a notification

Comment (jaalah-akamai) [reply packages/api-v4/src/account/types.ts:261]: We may not need `tax_id_valid` even though it's in the API (I will remove that and speak with API about doing the same). `tax_id_invalid` makes sense to me though, that's being returned if the validation of the tax id fails

Comment (bnussman-akamai) [reply packages/api-v4/src/account/types.ts:261]: Yeah agree. `tax_id_invalid` is perfect, just not sure how they were expecting `tax_id_valid` to behave

**Comment by jaalah-akamai** [comment]
I need to fix some invalidation issues with RQ - please hold

File: packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/ContactInformation.tsx [hunk 3]
```diff
@@ -74,10 +78,33 @@ const ContactInformation = (props: Props) => {
...
+  // to potentially clear the tax_id_invalid notification.
+  const handleTaxEventRefetching = () => {
+    const recentTaxIdValidEvent = events?.find((event) => {
+      return event.action === 'tax_id_valid';
+    });
+
+    if (recentTaxIdValidEvent) {
+      refetchNotifications();
+    }
+  };
```
Comment (jaalah-akamai) [inline packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/ContactInformation.tsx:98]: @bnussman-akamai This works, but I believe there could be old `tax_id_valid` that could give a false positive. I see we also have `useEventHandlers` or there may even be some other function that maybe does this better. Do you have a preference how this is done?

Comment (bnussman-akamai) [reply packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/ContactInformation.tsx:98]: I think `useEventHandlers` would be best just to keep with existing patterns

**Comment by jaalah-akamai** [comment]
Devenv has been having issues all day preventing me from e2e testing this - I will wait until things are back up and re-request reviews.

File: packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/ContactInformation.tsx [hunk 4]
```diff
@@ -74,10 +77,16 @@ const ContactInformation = (props: Props) => {
     setEditContactDrawerOpen,
   ] = React.useState<boolean>(false);
 
+  const { data: notifications } = useNotificationsQuery();
+
   const [focusEmail, setFocusEmail] = React.useState(false);
 
   const isChildUser = Boolean(profile?.user_type === 'child');
 
+  const invalidTaxId = notifications?.find((notification: any) => {
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/ContactInformation.tsx:86]:
```suggestion
  const invalidTaxId = notifications?.find((notification) => {
```

File: packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/ContactInformation.tsx [hunk 5]
```diff
@@ -74,10 +77,16 @@ const ContactInformation = (props: Props) => {
     setEditContactDrawerOpen,
   ] = React.useState<boolean>(false);
 
+  const { data: notifications } = useNotificationsQuery();
+
   const [focusEmail, setFocusEmail] = React.useState(false);
 
   const isChildUser = Boolean(profile?.user_type === 'child');
 
+  const invalidTaxId = notifications?.find((notification) => {
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/ContactInformation.tsx:86]: Optional:

```suggestion
  const invalidTaxIdNotification = notifications?.find((notification) => {
```

**Review by bnussman-akamai — APPROVED** [review]
React Query and event handler changes look great ✅

**Pull request merged**